data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/DefaultIndexManager.java

```diff
@@ -0,0 +1,15 @@
+package com.amazon.dataprepper.plugins.sink.opensearch.index;
+
+import com.amazon.dataprepper.plugins.sink.opensearch.OpenSearchSinkConfiguration;
+import com.amazon.dataprepper.plugins.sink.opensearch.index.ismpolicy.NoIsmPolicyManagement;
+import org.opensearch.client.RestHighLevelClient;
+
+public class DefaultIndexManager extends IndexManager {
```

Comment:
Yeah, if there is a DI framework like Guice, it may help eliminate children classes. There are other reasons I like have this inheritance: 
1. It allows us to clearly separately instantiate an IndexManager class of specific purpose. 
1. Because we have no.1, we can separate the unit test for different index types into different unit test classes. 
1. It would be clearer and time-saving for users of index manager to figure out which setup to use giving an index type. A programatic user of index manage (in my implementation, it is the index manager factory) also can have less to worry about and have a simpler implementation. 
1. It is easier for runtime debugging.  It's easier to figure out which index type the index manager instance is for

Reply:
I think keeping these implementations of `IndexManager` is still important. The OpenSearch plugin defines what each of these types is via an enum which is configured in the pipeline. This strong correspondence between the enum and a known setup for the index lends itself well to these classes (DefaultIndexManager, TraceAnalyticsManager).

Reply:
Let's revist this when we have injection. I believe all of this could be achieved with injection via a map of ENUMs to different implementations of an IndeManger. I believe this is a simplier design and reduces our reliance on inheritence. This does not limit our ability to test the different configuration options nor should it decrease runtime debugging or increase time for users.

Reply:
I believe the root problem is that these inherited classes are being exposed. From what I can tell of the code, we should be able to make them private static inner classes of the `IndexManagerFactory` class. At that point, it matters little whether it has its own inherited classes to expose them, or builds them all manually. It is just an implementation detail of `IndexManagerFactory`. I think a quick turnaround PR with this change would be appropriate.

Reply:
Sounds good. Good suggestion, David.  I am making the IndexManger
sub-classes private in the factory class. See: https://github.com/opensearch-project/data-prepper/pull/414 

I didn't make them static. I have not found being static is necessary.
Please let me know if you have a strong reason to make them static.